dace/sdfg/propagation.py

```diff
@@ -744,14 +744,12 @@ def propagate_states(sdfg: 'SDFG', concretize_dynamic_unbounded: bool = False) -
     :note: This operates on the SDFG in-place.
     """
 
-    if sdfg.using_explicit_control_flow:
-        # Avoid cyclic imports
-        from dace.transformation.pass_pipeline import Pipeline
-        from dace.transformation.passes.analysis import StatePropagation
-
-        state_prop_pipeline = Pipeline([StatePropagation()])
-        state_prop_pipeline.apply_pass(sdfg, {})
-        return
+    # Avoid cyclic imports
+    from dace.transformation.pass_pipeline import Pipeline
+    from dace.transformation.passes.analysis import StatePropagation
+
+    state_prop_pipeline = Pipeline([StatePropagation()])
+    state_prop_pipeline.apply_pass(sdfg, {})
```

Comment:
The original code has a return statement, so anything after this should not be executed.